sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -152,12 +156,36 @@ public InterceptorManager(String testName, Map<String, String> textReplacementRu
      * @throws UncheckedIOException An existing test session record could not be located or the data could not be
      * deserialized into an instance of {@link RecordedData}.
      * @throws NullPointerException If {@code testName} or {@code textReplacementRules} is {@code null}.
+     * @deprecated Use {@link #InterceptorManager(String, String, Map, boolean)} instead.
      */
+    @Deprecated
     public InterceptorManager(String testName, Map<String, String> textReplacementRules, boolean doNotRecord) {
+        this(testName, testName, textReplacementRules, doNotRecord);
+    }
+
+    /**
+     * Creates a new InterceptorManager that replays test session records. It takes a set of {@code
+     * textReplacementRules}, that can be used by {@link PlaybackClient} to replace values in a {@link
+     * NetworkCallRecord#getResponse()}.
+     *
+     * The test session records are read from: "<i>session-records/{@code testName}.json</i>"
+     *
+     * @param testName Name of the test.
+     * @param playbackRecordName Full name of the test including its iteration, used as the playback record name.
+     * @param textReplacementRules A set of rules to replace text in {@link NetworkCallRecord#getResponse()} when
+     * playing back network calls.
+     * @param doNotRecord Flag indicating whether network calls should be record or played back.
+     * @throws UncheckedIOException An existing test session record could not be located or the data could not be
+     * deserialized into an instance of {@link RecordedData}.
+     * @throws NullPointerException If {@code testName} or {@code textReplacementRules} is {@code null}.
+     */
+    public InterceptorManager(String testName, String playbackRecordName, Map<String, String> textReplacementRules,
```

Comment:
I'm used to adding new parameters at the end rather than inserting them in between because this is an "optional" parameter.

Reply:
Will move the new parameter to the end.